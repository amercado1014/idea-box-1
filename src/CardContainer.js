import React from 'react';
import Card from './Card';

const CardContainer = ({ideas, removeIdea}) => {
  const mappedIdeas = ideas.map((idea, index) => {
    return <Card 
             key={idea.id}
             id={idea.id}
             title={idea.title}
             description={idea.description}
             removeIdea={removeIdea}
           />
  })

  return(
    <div>
      {mappedIdeas}
    </div>
  )
}

export default CardContainer
