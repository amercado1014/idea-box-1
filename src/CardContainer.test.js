import CardContainer from './CardContainer'
import React from 'react'
import { shallow } from 'enzyme'

describe('CardContainer test', () => {
  it('should match the snapshot :D', () => {
    const mockIdeas = []
    const renderedApp = shallow(<CardContainer ideas={mockIdeas}/>);
    expect(renderedApp).toMatchSnapshot()
  })

  it('should render 3 cards', () => {
    const ideasToRender = [
      { id: 1, title: 'totes full', description: 'i ate alot of food' },
      { id: 2, title: 'why tho', description: 'tho why' },
      { id: 3, title: 'sheeiiitt', description: 'will is pretty kewl(really kewl)' }
    ]

    const expectCardLength = 3
    const renderedApp = shallow(<CardContainer ideas={ideasToRender}/>)
    //
    // expect(renderedApp.find('Card').length).toEqual(expectCardLength);
    expect(renderedApp).toMatchSnapshot()
  });
})
