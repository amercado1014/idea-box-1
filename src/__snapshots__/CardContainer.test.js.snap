// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer test should match the snapshot :D 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    ideas={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
    },
    "ref": null,
    "rendered": Array [],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
      },
      "ref": null,
      "rendered": Array [],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`CardContainer test should render 3 cards 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    ideas={
        Array [
            Object {
              "description": "i ate alot of food",
              "id": 1,
              "title": "totes full",
            },
            Object {
              "description": "tho why",
              "id": 2,
              "title": "why tho",
            },
            Object {
              "description": "will is pretty kewl(really kewl)",
              "id": 3,
              "title": "sheeiiitt",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          description="i ate alot of food"
          id={1}
          removeIdea={undefined}
          title="totes full"
/>,
        <Card
          description="tho why"
          id={2}
          removeIdea={undefined}
          title="why tho"
/>,
        <Card
          description="will is pretty kewl(really kewl)"
          id={3}
          removeIdea={undefined}
          title="sheeiiitt"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "description": "i ate alot of food",
          "id": 1,
          "removeIdea": undefined,
          "title": "totes full",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "description": "tho why",
          "id": 2,
          "removeIdea": undefined,
          "title": "why tho",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "function",
        "props": Object {
          "description": "will is pretty kewl(really kewl)",
          "id": 3,
          "removeIdea": undefined,
          "title": "sheeiiitt",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            description="i ate alot of food"
            id={1}
            removeIdea={undefined}
            title="totes full"
/>,
          <Card
            description="tho why"
            id={2}
            removeIdea={undefined}
            title="why tho"
/>,
          <Card
            description="will is pretty kewl(really kewl)"
            id={3}
            removeIdea={undefined}
            title="sheeiiitt"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "description": "i ate alot of food",
            "id": 1,
            "removeIdea": undefined,
            "title": "totes full",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "description": "tho why",
            "id": 2,
            "removeIdea": undefined,
            "title": "why tho",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "description": "will is pretty kewl(really kewl)",
            "id": 3,
            "removeIdea": undefined,
            "title": "sheeiiitt",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
